packages/core-react/src/hooks/useEagerConnect.ts

```diff
@@ -1,35 +1,44 @@
-import { Chain } from 'wagmi';
-import { connect, disconnect } from 'wagmi/actions';
-import type { ConnectResult, Connector } from '@wagmi/core';
 import { useCallback } from 'react';
-import { WalletAdapterData } from '@reef-knot/types';
-import { getUnsupportedChainError } from '../helpers/getUnsupportedChainError';
-import { checkTermsAccepted } from '../helpers/checkTermsAccepted';
+import { useConfig } from 'wagmi';
 import { useReefKnotContext } from './useReefKnotContext';
-import { ReefKnotModalContextValue } from '../context';
 import { useReefKnotModal } from './useReefKnotModal';
 
+import { connect, disconnect } from 'wagmi/actions';
+import { getUnsupportedChainError } from '../helpers/getUnsupportedChainError';
+import { checkTermsAccepted } from '../helpers/checkTermsAccepted';
+
+import type { WalletConnectorData } from '@reef-knot/types';
+import type { Chain } from 'wagmi/chains';
+import type { Config, Connector, ConnectReturnType } from '@wagmi/core';
+import type { ReefKnotModalContextValue } from '../context';
+
+type ConnectResult = ConnectReturnType;
+
 const connectToAdapter = async (
+  config: Config,
   connector: Connector,
-  supportedChains: Chain[],
+  supportedChains: readonly [Chain, ...Chain[]],
 ) => {
-  const connectResult = await connect({ connector });
+  const connectResult = await connect(config, { connector });
 
-  if (connectResult?.chain.unsupported) {
+  if (
+    connectResult &&
+    supportedChains.findIndex(({ id }) => id === connectResult.chainId) === -1
+  ) {
     // No errors during connection, but the chain is unsupported.
     // This case is considered as error for now, and we explicitly call disconnect() here.
     // This logic comes from previously used web3-react connection logic, which wasn't reworked yet after web3-react removal.
     // web3-react logic was: if a chain is unsupported – break the connection, throw an error
     // wagmi logic is: if a chain is unsupported – connect anyway, without errors, set `chain.unsupported` flag to true.
```

Comment:
This comment looks obsolete now, since there is no `chain.unsupported` flag anymore

Reply:
I have checked this scenario and it is still presented as before in wagmi — if no additional checks performed it just connects to unsupported network. There is just no flag presented now. So I will made a correction for the comment, but the case is still here.